lib/model/message.dart

```diff
@@ -0,0 +1,167 @@
+import '../api/model/events.dart';
+import '../api/model/model.dart';
+import 'message_list.dart';
+
+/// The portion of [PerAccountStore] for messages and message lists.
+mixin MessageStore {
+  /// All known messages, indexed by [Message.id].
+  Map<int, Message> get messages;
+
+  void registerMessageList(MessageListView view);
+  void unregisterMessageList(MessageListView view);
+
+  /// Reconcile a batch of just-fetched messages with the store,
+  /// mutating the list.
+  ///
+  /// This is called after a [getMessages] request to report the result
+  /// to the store.
+  ///
+  /// The list's length will not change, but some entries may be replaced
+  /// by a different [Message] object with the same [Message.id].
+  /// All [Message] objects in the resulting list will be present in [messages].
```

Comment:
Hmm, this can probably be clearer. :slightly_smiling_face:

The "resulting list" is indeed the function's parameter, `messages`.

But by `[messages]` here I meant the *field* (or getter), aka `this.messages`. (I actually didn't notice when I wrote this that there was a name collision.)

So to unpack: when the function is done, all the `Message` objects then found in the list will also be present in the `messages` map on this message store.

Reply:
Cool, makes sense. So maybe we just change the last word in that sentence from `[messages]` to `[this.messages]`?

Reply:
Yeah, SGTM